packages/core/src/StateNode.ts

```diff
@@ -414,7 +414,7 @@ class StateNode<
         return toInvokeDefinition({
           ...invokeConfig,
 
-          id: invokeConfig.id || (invokeConfig.src as string),
+          id: invokeConfig.id || `${this.id}:invocation[${i}]`,
```

Comment:
I believe that we still have the same problem in other branches of this `.map`:
1. in the `isMachine(invokeConfig)` branch we should always use the auto-generated id as [reusing `invokeConfig.id`](https://github.com/statelyai/xstate/blob/4c2fdbdcefafa5db71f150506814f49ef70e0ee6/packages/core/src/StateNode.ts#L411) is dangerous for the very same reason, this would also match the behavior of the branch that checks `isMachine(invokeConfig.src) || isFunction(invokeConfig.src)` as in there the [`invokeSrc` is always autogenerated](https://github.com/statelyai/xstate/blob/4c2fdbdcefafa5db71f150506814f49ef70e0ee6/packages/core/src/StateNode.ts#L421)
2. the else branch for parametrized invocations is also [reusing `invokeConfig.src.type`](https://github.com/statelyai/xstate/blob/4c2fdbdcefafa5db71f150506814f49ef70e0ee6/packages/core/src/StateNode.ts#L436) which is even more close to the bug already fixed in this PR as this is always~ a string referencing an actor definition that can be looked up in the `implementations` argument

Reply:
Just a reminder about this comment that blocks merging this PR. If you want I can add failing tests for those tomorrow and just patch those lines as those are really one-liners.

Reply:
Added here: 9ad6b8fd7f09cc70cc1e377628f27c83aff835fc

Reply:
The first described scenario was not fixed in this commit, I've added a failing test case and fixed this issue in: https://github.com/statelyai/xstate/pull/2915